pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -352,7 +360,15 @@ public static void start(String[] args, SendGridSupplier sendGridSupplier) {
                                 TransactionWrapper.useTxn(TransactionWrapper.DB.HOUSEKEEPING, housekeepingHandle -> {
                                     JdbiEvent jdbiEvent = housekeepingHandle.attach(JdbiEvent.class);
                                     JdbiMessage jdbiMessage = housekeepingHandle.attach(JdbiMessage.class);
-                                    if (jdbiEvent.shouldHandleEvent(
+                                    if (pendingEvent.getActionType() == EventActionType.ACTIVITY_INSTANCE_CREATION) {
+                                        Optional<EventConfigurationDto> eventConfOptDto =
+                                                eventDao.getEventConfigurationDtoById(pendingEvent.getEventConfigurationId());
+                                        if (!eventConfOptDto.isPresent()) {
+                                            LOG.error("No event configuration found for ID: {} . skipping queued event : {} ",
+                                                    pendingEvent.getEventConfigurationId(), pendingEvent.getQueuedEventId());
+                                        }
+                                        createActivityInstance(apisHandle, eventConfOptDto.get(), pendingEvent);
```

Comment:
Can you move this block of code to _outside_ of `TransactionWrapper` call? So that we don't unnecessarily open a `DB.HOUSEKEEPING` database connection?